L1Trigger/L1THGCal/src/backend/HGCalHistoSeedingImpl.cc

```diff
@@ -166,16 +174,21 @@ HGCalHistoSeedingImpl::Histogram HGCalHistoSeedingImpl::fillSmoothPhiHistoCluste
   for (int z_side : {-1, 1}) {
     for (unsigned bin1 = 0; bin1 < nBins1_; bin1++) {
       int nBinsSide = (binSums[bin1] - 1) / 2;
-      float R1 = kROverZMin_ + bin1 * (kROverZMax_ - kROverZMin_) / nBins1_;
-      float R2 = R1 + ((kROverZMax_ - kROverZMin_) / nBins1_);
       double area =
-          ((M_PI * (pow(R2, 2) - pow(R1, 2))) / nBins2_) *
           (1 +
            2.0 *
                (1 -
                 pow(0.5,
                     nBinsSide)));  // Takes into account different area of bins in different R-rings + sum of quadratic weights used
 
+      if (seeds_norm_by_area_) {
+        float R1 = kROverZMin_ + bin1 * (kROverZMax_ - kROverZMin_) / nBins1_;
+        float R2 = R1 + ((kROverZMax_ - kROverZMin_) / nBins1_);
+        area = area * ((M_PI * (pow(R2, 2) - pow(R1, 2))) / nBins2_);
```

Comment:
Again, not relevant but for the aesthetics:
```suggestion
        float R2 = R1 + (kROverZMax_ - kROverZMin_) / nBins1_;
        area = area * M_PI * (pow(R2, 2) - pow(R1, 2)) / nBins2_;
```